src/main/java/com/farao_community/farao/dichotomy/rao/RaoStepResult.java

```diff
@@ -6,33 +6,29 @@
  */
 package com.farao_community.farao.dichotomy.rao;
 
-import com.farao_community.farao.data.crac_api.Crac;
-import com.farao_community.farao.data.crac_result_extensions.CracResult;
-import com.farao_community.farao.data.crac_result_extensions.CracResultExtension;
+import com.farao_community.farao.data.rao_result_api.ComputationStatus;
+import com.farao_community.farao.data.rao_result_api.OptimizationState;
+import com.farao_community.farao.data.rao_result_api.RaoResult;
 import com.farao_community.farao.dichotomy.StepResult;
-import com.farao_community.farao.rao_api.RaoResult;
 
 /**
  * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
  */
 public class RaoStepResult implements StepResult {
     private final double stepValue;
-    private final Crac crac;
     private final RaoResult raoResult;
 
-    public RaoStepResult(double stepValue, Crac crac, RaoResult raoResult) {
+    public RaoStepResult(double stepValue, RaoResult raoResult) {
         this.stepValue = stepValue;
-        this.crac = crac;
         this.raoResult = raoResult;
     }
 
     @Override
     public boolean isSecure() {
-        if (!raoResult.isSuccessful() || crac.getExtension(CracResultExtension.class) == null) {
+        if (raoResult.getComputationStatus() == ComputationStatus.FAILURE) {
             return false;
         }
-        CracResult cracResult = crac.getExtension(CracResultExtension.class).getVariant(raoResult.getPostOptimVariantId());
-        return cracResult.getNetworkSecurityStatus() == CracResult.NetworkSecurityStatus.SECURED;
+        return raoResult.getFunctionalCost(OptimizationState.AFTER_CRA) > 0;
```

Comment:
Wouldn't it be useful to add the isSecure method directly in raoResult API ? It seems to be useful business wise and only use information from raoResult object here.

Reply:
We stated with FARAO team that what is behind functional cost depends on the objective function definition and that we could not state about network security status just from functional cost in FARAO. So it's the responsibility of the user if he wants to state about network security status to implement it.